src/syscheckd/registry/registry.c

```diff
@@ -509,206 +534,154 @@ void fim_registry_free_entry(fim_entry *entry) {
- */
-void fim_registry_process_key_delete_event(fdb_t *fim_sql,
-                                           fim_entry *data,
-                                           pthread_mutex_t *mutex,
-                                           void *_alert,
-                                           void *_ev_mode,
-                                           void *_w_evt) {
-    int alert = *(int *)_alert;
-    fim_event_mode event_mode = *(fim_event_mode *)_ev_mode;
-    fim_tmp_file *file;
-    registry_t *configuration;
-    int result;
+    cJSON *json_event = cJSON_CreateObject();
+    cJSON_AddStringToObject(json_event, "type", "event");
 
-    configuration = fim_registry_configuration(data->registry_entry.key->path, data->registry_entry.key->arch);
-    if (configuration == NULL) {
-        return;
-    }
+    cJSON *data = cJSON_CreateObject();
+    cJSON_AddItemToObject(json_event, "data", data);
 
-    if (alert) {
-        cJSON *json_event = fim_registry_event(data, NULL, configuration, event_mode, FIM_DELETE, NULL, NULL);
+    cJSON_AddStringToObject(data, "path", cJSON_GetObjectItem(dbsync_event, "path")->valuestring);
+    cJSON_AddNumberToObject(data, "version", 2.0);
+    cJSON_AddStringToObject(data, "mode", FIM_EVENT_MODE[mode]);
+    cJSON_AddStringToObject(data, "type", FIM_EVENT_TYPE_ARRAY[evt_data->type]);
+    cJSON_AddStringToObject(data, "arch", cJSON_GetObjectItem(dbsync_event, "arch")->valuestring);
+    cJSON_AddStringToObject(data, "value_name", cJSON_GetObjectItem(dbsync_event, "name")->valuestring);
+    cJSON_AddItemToObject(data, "attributes", fim_registry_value_attributes_json(dbsync_event,
+                          configuration, data_type));
 
-        if (json_event) {
-            send_syscheck_msg(json_event);
-            cJSON_Delete(json_event);
-        }
+    if (diff != NULL) {
+        cJSON_AddStringToObject(data, "content_changes", diff);
     }
-    /* DEPRECATED CODE
-    result = fim_db_get_values_from_registry_key(fim_sql, &file, syscheck.database_store, data->registry_entry.key->id);
 
-    if (result == FIMDB_OK && file && file->elements) {
-        fim_db_process_read_registry_data_file(fim_sql, file, mutex, fim_registry_process_value_delete_event,
-                                               syscheck.database_store, _alert, _ev_mode, _w_evt);
+
+ /* cJSON_AddStringToObject(data, "timestamp",  cJSON_GetObjectItem(dbsync_event, "last_event")->valuestring);
+    if (old_data != NULL && old_data->registry_entry.value != NULL) {
+        cJSON_AddItemToObject(data, "changed_attributes", changed_attributes);
+        cJSON_AddItemToObject(data, "old_attributes",
+                              fim_registry_value_attributes_json(old_data->registry_entry.value, configuration));
     }
 
-    fim_db_remove_registry_key(fim_sql, data);
+    if (diff != NULL) {
+        cJSON_AddStringToObject(data, "content_changes", diff);
+    }*/
```

Comment:
If this code is useless, you can delete it

Reply:
Done in [commit](https://github.com/wazuh/wazuh/pull/11239/commits/11a6215594b89425c4469067a26e23d1a1a7f5df)